x-pack/legacy/plugins/ml/public/application/data_frame_analytics/pages/analytics_exploration/components/classification_exploration/results_table.tsx

```diff
@@ -0,0 +1,482 @@
+      pageSizeOptions: PAGE_SIZE_OPTIONS,
+      hidePerPageOptions: false,
+    };
+
+    const onQueryChange = ({ query, error }: { query: QueryType; error: any }) => {
+      if (error) {
+        setSearchError(error.message);
+      } else {
+        try {
+          const esQueryDsl = Query.toESQuery(query);
+          setSearchQuery(esQueryDsl);
+          setSearchString(query.text);
+          setSearchError(undefined);
+          // set query for use in evaluate panel
+          setEvaluateSearchQuery(esQueryDsl);
+        } catch (e) {
+          setSearchError(e.toString());
+        }
+      }
+    };
+
+    const search = {
+      onChange: onQueryChange,
+      defaultQuery: searchString,
+      box: {
+        incremental: false,
+        placeholder: i18n.translate(
+          'xpack.ml.dataframe.analytics.regressionExploration.searchBoxPlaceholder',
+          {
+            defaultMessage: 'E.g. avg>0.5',
+          }
+        ),
+      },
+      filters: [
+        {
+          type: 'field_value_toggle_group',
+          field: `${jobConfig.dest.results_field}.is_training`,
+          items: [
+            {
+              value: false,
+              name: i18n.translate(
+                'xpack.ml.dataframe.analytics.regressionExploration.isTestingLabel',
+                {
+                  defaultMessage: 'Testing',
+                }
+              ),
+            },
+            {
+              value: true,
+              name: i18n.translate(
+                'xpack.ml.dataframe.analytics.regressionExploration.isTrainingLabel',
+                {
+                  defaultMessage: 'Training',
+                }
+              ),
+            },
+          ],
+        },
+      ],
+    };
+
+    if (jobConfig === undefined) {
+      return null;
+    }
+    // if it's a searchBar syntax error leave the table visible so they can try again
+    if (status === INDEX_STATUS.ERROR && !errorMessage.includes('parsing_exception')) {
+      return (
+        <EuiPanel grow={false}>
+          <EuiFlexGroup gutterSize="s">
+            <EuiFlexItem grow={false}>
+              <ExplorationTitle jobId={jobConfig.id} />
+            </EuiFlexItem>
+            <EuiFlexItem grow={false}>
+              <span>{getTaskStateBadge(jobStatus)}</span>
+            </EuiFlexItem>
+          </EuiFlexGroup>
+          <EuiCallOut
+            title={i18n.translate('xpack.ml.dataframe.analytics.regressionExploration.indexError', {
+              defaultMessage: 'An error occurred loading the index data.',
+            })}
+            color="danger"
+            iconType="cross"
+          >
+            <p>{errorMessage}</p>
+          </EuiCallOut>
+        </EuiPanel>
+      );
+    }
+
+    const tableError =
+      status === INDEX_STATUS.ERROR && errorMessage.includes('parsing_exception')
+        ? errorMessage
+        : searchError;
+
+    const MlInMemoryTableBasic = mlInMemoryTableBasicFactory<TableItem>();
```

Comment:
This line should be moved out of the component on the top level of the module, see https://github.com/elastic/kibana/pull/52486 for reference.

Reply:
Definitely a necessary change. Updating for classification and regression results views. 👌